packages/transformers/jsonld/src/JSONLDTransformer.js

```diff
@@ -0,0 +1,94 @@
+// @flow
+
+import {Transformer} from '@parcel/plugin';
+import json5 from 'json5';
+
+// A list of all attributes in a schema that may produce a dependency
+// Based on https://schema.org/ImageObject
+// Section "Instances of ImageObject may appear as values for the following properties"
+const SCHEMA_ATTRS = [
+  'logo',
+  'photo',
+  'image',
+  'thumbnail',
+  'screenshot',
+  'primaryImageOfPage',
+  'embedUrl',
+  'thumbnailUrl',
+  'video',
+  'contentUrl',
+  'url'
+];
+
+export default new Transformer({
+  async transform({ asset, config, logger, options, resolve }) {
+    
+    let rawCode = await asset.getCode();
+    // allowing any recieved jsonld to be in json5 format
+    let jsonCode = json5.parse(rawCode);
+    
+    let parser = new JSONLDParser(asset);
+    parser.parse(jsonCode);
+    
+    // should these be used to tell the transformer that the output is meant to be placed back into the index.html file?
+    //asset.isInline = true;
+    //asset.meta.inlineType = 'string';
```

Comment:
so locally I made the change that @DeMoorJasper suggested [here](https://github.com/parcel-bundler/parcel/pull/4397#discussion_r399685120) . After that, the built index.html file has `<div></div>` inside the script tag where the jsonld object should be. I am unsure why its putting a div in there and not the transformed jsonld object (which really should be the original jsonld object with updated asset paths).

After doing some research in the code, I was thinking that these options might be close to what I need? I tried them out and the output remains the same. Not sure how to proceed to get it to write the jsonld object into the index.html file.

Reply:
take a look at the [result][1] of the unit test I wrote, it shows that output being the div instead of the expected output of the json object

[1]: https://dev.azure.com/devongovett/devongovett/_build/results?buildId=8730&view=logs&j=3dbdc9c4-c12f-5bc5-3f55-0178d7f6c002&t=59bda16f-6692-523f-b778-442dd3cca32c&l=761

Reply:
I think I have it figured out in my latest commit --- I changed `asset.type="jsonld"` to `asset.type="html"`

Reply:
Setting it to `html` is definitely not what we want. It should be `json`, but we'll have to figure out how to to always convert JSON assets into JS modules for that.